contracts/token/ERC1155/extensions/ERC1155Supply.sol

```diff
@@ -66,11 +66,8 @@ abstract contract ERC1155Supply is ERC1155 {
             for (uint256 i = 0; i < ids.length; ++i) {
                 uint256 id = ids[i];
                 uint256 amount = amounts[i];
-                uint256 supply = _totalSupply[id];
-                require(supply >= amount, "ERC1155: burn amount exceeds totalSupply");
+                _totalSupply[id] -= amount;
```

Comment:
Is that ok ? 

If the amount is greater then the totalSupply, this will revert without a message, and we won't get any feedback.
Since this is done BEFORE the super call, we won't have the "insufficient balance" error thrown.

Maybe we want this operation to be unchecked, knowing that any overflow here would necessarily be the result of an amount that is above the balance, and thus rely on the super call to trigger the error.

Reply:
I don't have a strong opinion and I agreed with @frangio [here](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/4261#discussion_r1214592540) on doing it this way.

Still, I think the take of using the `InsufficientBalance` error is an excellent reason to keep it. I'm leaving this open but my standing is that both approaches have advantages and disadvantages. A summary would be:

**Keeping ERC1155InsufficientBalance**: Better error information, more overhead.
**Relying on Solidity checked arithmetic**: Less information, less overhead.

Reply:
Bumping here my suggestion was that we should be caling `_update` before we update the supply. This means that the error due to insufficient balance will be emitted by `_update`, and the checked arithmetic is used for unexpected errors due to out of sync balances and supply.

Reply:
Thanks for bumping, I think it makes sense. Just changed the order as suggested.

Reply:
So what do we do here: I'd go the ERC1155InsufficientBalance way for now, and possibly change that if we refactor the _update function in a further PR.